From: David Grayson <davidegrayson@gmail.com>
Subject: [PATCH] qtwebkit: Disambiguate call to WTF::StringBuilder::append(int)

The call to the overloaded method WTF::StringBuilder::append(int) was ambiguous.
The compiler could not decide whether to convert to UChar, LChar, char or UChar32.
--- qt-everywhere-opensource-src-5.5.1.orig/qtwebkit/Source/WebCore/rendering/RenderText.cpp
+++ qt-everywhere-opensource-src-5.5.1/qtwebkit/Source/WebCore/rendering/RenderText.cpp
@@ -126,7 +126,7 @@
     int32_t startOfWord = textBreakFirst(boundary);
     for (endOfWord = textBreakNext(boundary); endOfWord != TextBreakDone; startOfWord = endOfWord, endOfWord = textBreakNext(boundary)) {
         if (startOfWord) // Ignore first char of previous string
-            result.append(stringImpl[startOfWord - 1] == noBreakSpace ? noBreakSpace : toTitleCase(stringWithPrevious[startOfWord]));
+            result.append((UChar)(stringImpl[startOfWord - 1] == noBreakSpace ? noBreakSpace : toTitleCase(stringWithPrevious[startOfWord])));
         for (int i = startOfWord + 1; i < endOfWord; i++)
             result.append(stringImpl[i - 1]);
     }
